### PR TITLE
Remove deprecation warnings

### LINKED ISSUE #3190: Play 2.6 Follow-Up: tackle remaining deprecation warnings

### PR DESCRIPTION
**Pull request by rschwanhold**
### Steps to test:
- run `./clean`
- run `yarn start`
- there should not be any deprecation warnings of play

### Issues:
- fixes #3190
------
- ~~[ ] Updated [changelog](../blob/master/CHANGELOG.md#unreleased)~~
- ~~[ ] Updated [migration guide](../blob/master/MIGRATIONS.md#unreleased) if applicable~~
- ~~[ ] Updated [documentation](../blob/master/docs) if applicable~~
- [x] Needs datastore update after deployment
- [x] Ready for review

### CODE REVIEW

**Comment by rschwanhold** [comment]
@jstriebel I am struggeling to remove one of the deprecation warnings. The warning is: `  [warn] /home/robert/Dokumente/Arbeit/thirdTry/webknossos/webknossos-datastore/app/com/scalableminds/webknossos/datastore/storage/TemporaryStore.scala:12: method apply in object Agent is deprecated: Agents are deprecated and scheduled for removal in the next major version, use Actors instead.
  [warn]   lazy val ts = Agent[Map[K, V]](Map())`
As you can see here (https://doc.akka.io/docs/akka/current/project/migration-guide-2.4.x-2.5.x.html?language=scala#agents) Agents are now deprecated and they recommend to use Actors instead. But there is no streight forward way to convert an Agent into an Actor.
I started to create an Actor in my last commit, but I think this is not how it should be. How would you try to solve it?
(About Actors: https://www.playframework.com/documentation/2.6.x/ScalaAkka)

**Comment by fm3** [comment]
@rschwanhold @jstriebel Akka Actors seem overly complicated for this use case. I think using scala synchronized should yield the same behavior as before. (blocking the executing thread, but not other requests). Does the version I just pushed look ok to you?

**Comment by jstriebel** [comment]
@fm3 Good point, simply not using akka here is the best option. Instead of doing the synchronization ourselves we could directly use a concurrent map](https://docs.scala-lang.org/overviews/collections/concrete-mutable-collection-classes.html#concurrent-maps).

File: app/controllers/DataSetController.scala [hunk 1]
```diff
@@ -33,7 +32,7 @@ class DataSetController @Inject()(userService: UserService,
                                   teamDAO: TeamDAO,
                                   dataSetDAO: DataSetDAO,
                                   sil: Silhouette[WkEnv],
-                                  cache: CacheApi)
+                                  cache: SyncCacheApi)
```
Comment (rschwanhold) [inline app/controllers/DataSetController.scala]: We have the choice between `SyncCacheApi` and `AsyncCacheApi`

**Pull request merged**